src/components/notifications/ui-notification-boundary.tsx

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { PropsWithChildren } from 'react'
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react'
+import { UiNotifications } from './ui-notifications'
+import type { DispatchOptions, UiNotification } from './types'
+import { DateTime } from 'luxon'
+import type { TOptions } from 'i18next'
+import { t } from 'i18next'
+import { Logger } from '../../utils/logger'
+import { v4 as uuid } from 'uuid'
+
+const log = new Logger('Notifications')
+
+interface UiNotificationContext {
+  dispatchUiNotification: (
+    titleI18nKey: string,
+    contentI18nKey: string,
+    { icon, durationInSecond, buttons, contentI18nOptions, titleI18nOptions }: Partial<DispatchOptions>
+  ) => void
+
+  showErrorNotification: (messageI18nKey: string, messageI18nOptions?: TOptions | string) => (error: Error) => void
+
+  dismissNotification: (notificationUuid: string) => void
+}
+
+export const useUiNotifications: () => UiNotificationContext = () => {
+  const communicatorFromContext = useContext(uiNotificationContext)
+  if (!communicatorFromContext) {
+    throw new Error('No ui notifications')
+  }
+  return communicatorFromContext
+}
+
+export const DEFAULT_DURATION_IN_SECONDS = 10
+const uiNotificationContext = createContext<UiNotificationContext | undefined>(undefined)
+
+export const UiNotificationBoundary: React.FC<PropsWithChildren> = ({ children }) => {
+  const [uiNotifications, setUiNotifications] = useState<UiNotification[]>([])
+
+  const dispatchUiNotification = useCallback(
+    (
+      titleI18nKey: string,
+      contentI18nKey: string,
+      { icon, durationInSecond, buttons, contentI18nOptions, titleI18nOptions }: Partial<DispatchOptions>
```

Comment:
As we have this set of options in multiple places, does it make sense to create an extra type for it (e.g with Pick<DispatchOptions>...)?

Reply:
I removed it from the interface because it doesn't make sense there. But I left it hear for deconstruction reasons.

Reply:
ok